ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/activity.component.ts

```diff
@@ -400,4 +405,30 @@ export class ActivityComponent extends BaseActivityComponent implements OnInit,
         this.isAdminEditing = adminEditing;
         this.changeRef.detectChanges();
     }
+
+    private addServerValidationMessagesToViolatedQuestions(error: HttpErrorResponse): void {
+        if (!error?.error?.violations) {
+            this.logger.logError('ActivityComponent', 'There was an error during submission (a violated question handler):', error);
+            return;
+        }
+        const questionBlocks = this.getQuestionBlocks();
+        for (const violation of error.error.violations) {
+            const violatedQuestions = questionBlocks.filter(qBlock => qBlock.stableId === violation.stableId);
+            this.handleViolation(violatedQuestions, violation.rules);
+        }
+    }
+
+    private handleViolation(violatedQuestions: AbstractActivityQuestionBlock[], validationRules = []): void {
+        for (const violatedQuestion of violatedQuestions) {
+            violatedQuestion.serverValidationMessages = [];
+            validationRules.map(rule => rule.message)
+                .filter(Boolean)
```

Comment:
Not sure what is going with this filter, could we make it more readable?

Reply:
It is kind of an alias for ` filter(x => !!x) `.

The **Boolean** global reference points to the constructor function which returns a boolean value (true / false) from the first argument. 
E.g. `Boolean('some text' | 123 | [] | {}) => true ; Boolean(null | undefined | 0 | '') => false`

Thus:
`.filter(Boolean)` === `.filter(x => Boolean(x))` === `.filter(x => !!x)` 


[docs link](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#:~:text=to%20convert%20a%20non%2Dboolean%20value%20to%20a%20boolean%20value.%20To%20perform%20this%20task%2C%20instead%2C%20use%20Boolean%20as%20a%20function%2C%20or%20a%20double%20NOT%20operator%3A)